torch/_dynamo/variables/functions.py

```diff
@@ -639,3 +639,77 @@ def as_python_constant(self):
                 *[arg.as_python_constant for arg in self.args],
                 **{k: v.as_python_constant() for k, v in self.keywords.items()},
             )
+
+
+class TritonKernelVariable(VariableTracker):
+    def __init__(self, kernel, grid, **kwargs):
+        super().__init__(**kwargs)
+        self.kernel = kernel
+        self.grid = grid
+
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        from .dicts import ConstDictVariable
+        from .lists import BaseListVariable
+
+        grid = self.grid
+
+        assert grid is not None
+
+        # If the grid is a function, then lets execute it and convert it to
+        # a list
+        if isinstance(grid, (NestedUserFunctionVariable, UserFunctionVariable)):
+            # Populate the special "meta" argument to call the grid function
+            d = {**dict(zip(self.kernel.arg_names, args)), **kwargs}
+            meta = ConstDictVariable(d, dict)
+            grid = grid.call_function(tx, [meta], {})
+
+        # Now, the grid must be a list either originally or through above
+        # modification
+        if isinstance(grid, BaseListVariable):
+            grid = grid.as_proxy()
+        else:
+            unimplemented(f"grid for the triton kernel is {type(grid)}")
+
+        from torch._higher_order_ops.triton_kernel_wrap import (
+            triton_kernel_wrapper_mutation,
+        )
+
+        proxied_args, proxied_kwargs = proxy_args_kwargs(args, kwargs)
+        fn = functools.partial(triton_kernel_wrapper_mutation, kernel=self.kernel)
+        # FX graph needs __name__ and __module__ attributes
+        fn.__name__ = triton_kernel_wrapper_mutation.__name__
+        if not hasattr(fn, "__module__"):
+            # Super hacky but on AMD __module__ is not set
+            fn.__module__ = "itertools"
```

Comment:
I'm not sure this is an amd thing, this might be a python version thing (locally for me functools.partial doesn't return something with a `.__module__`). 

Also it would be better to set the `__module__` to triton_kernel_wrapper_mutation's `.__module__`: IIRC these two things are responsible for how the function displays in the dynamo graph